### PR TITLE
#17 | Create VeeProgressCircular component.

### LINKED ISSUE #17: Implement VeeProgressCircular component
**Is your feature request related to a problem? Please describe.**
/

**Describe the solution you'd like**
We need to implement VeeProgressCircular component to represent a progress circular.

**Describe alternatives you've considered**
/

**Additional context**
Please follow the documentation: https://v2.vuetifyjs.com/en/components/progress-circular/ and https://v2.vuetifyjs.com/en/api/v-progress-circular/

### PR DESCRIPTION
**Pull request by andreadelfante**
# Pull Request
Compile all the information below to send a legitimate pull request to the repository!

### This PR is labeled as:
*Choose only one and remember to select the label even on the right.*

- [x] Feature
- [ ] Bug
- [ ] Other

### This PR has the following linked issues:
- Closes #17;

### CODE REVIEW

File: packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+  >
+    <svg
+      :style="veeSvgStyle"
+      xmlns="http://www.w3.org/2000/svg"
+      :viewBox="`${veeViewBoxSize} ${veeViewBoxSize} ${2 * veeViewBoxSize} ${
+        2 * veeViewBoxSize
+      }`"
+    >
+      <circle
+        v-if="!indeterminate"
```
Comment (Hrabal) [inline packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue:18]: Maybe is worth to add an `underlay` prop?

File: packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        :cy="veeViewBoxSize * 2"
+        :r="radius"
+        :stroke-width="veeStrokeWidth"
+        :stroke-dasharray="veeCircumference"
+        :stroke-dashoffset="veeStrokeDashOffset"
+      />
+    </svg>
+
+    <div
+      v-if="!this.indeterminate"
```
Comment (Hrabal) [inline packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue:42]: Is `this.` correct?

File: packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+.vee-progress-circular-overlay {
+  color: v-bind('veeColor');
+  stroke: currentColor;
+  transition: all 0.6s ease-in-out;
+  z-index: 2;
+}
+
+.vee-progress-circular-overlay-indeterminate {
+  animation: vee-progress-circular-dash 1.4s linear infinite;
+  stroke-linecap: round;
```
Comment (Hrabal) [inline packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue:230]: I'd leave the overlay style the same regardless of `indeterminate` value.. (maybe a round prop too?)

File: packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    },
+
+    value: {
+      type: [Number, String],
+      default: 0,
+    },
+
+    width: {
+      type: [Number, String],
+      default: 4,
```
Comment (Hrabal) [inline packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue:83]: I'd make the default width calculated as "size / 8".

File: packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+    },
+
+    veeNormalizedValue(): number {
+      return this.indeterminate
+        ? 0
+        : Math.min(100, Math.max(0, Number(this.value)));
+    },
+
+    veeStrokeWidth(): number {
+      return (Number(this.width) / +this.size) * this.veeViewBoxSize * 2;
```
Comment (walter-hlpy) [inline packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue:125]:
```suggestion
      return (Number(this.width) / + Number(this.size)) * this.veeViewBoxSize * 2;
```

File: packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+    veeColor(): string {
+      return this.$crumbs.themesManager.getColorRef(this.color);
+    },
+
+    veeColorUnderlay(): string {
+      return this.$crumbs.themesManager.getColorRef('greyLighten3');
+    },
+
+    veeViewBoxSize(): number {
+      return this.radius / (1 - Number(this.width) / +this.size);
```
Comment (walter-hlpy) [inline packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue:141]:
```suggestion
      return this.radius / (1 - Number(this.width) / + Number(this.size));
```

File: packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/stories.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  }),
+
+  argTypes: {
+    color: {
+      control: { type: 'text' },
+      description: `Sets the color of the progress indicator. Applies specified color to the control - it can be the name of theme color (for example ${documentCode('primary')}) or hex color.`,
+    },
+
+    size: {
+      control: { type: 'number' },
```
Comment (walter-hlpy) [inline packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/stories.ts:24]: This accept also a string

File: packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/stories.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+      description: 'Constantly animates, use when loading progress is unknown.',
+    },
+
+    value: {
+      control: { type: 'number' },
+      description: `The percentage value for current progress ${documentRange(0, 100)}.`,
+    },
+
+    width: {
+      control: { type: 'number' },
```
Comment (walter-hlpy) [inline packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/stories.ts:39]: This accept also a string

File: packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/stories.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+      description: 'Sets the diameter of the circle in pixels.',
+    },
+
+    indeterminate: {
+      control: { type: 'boolean' },
+      description: 'Constantly animates, use when loading progress is unknown.',
+    },
+
+    value: {
+      control: { type: 'number' },
```
Comment (walter-hlpy) [inline packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/stories.ts:34]: This accept also a string

**Review by walter-hlpy — CHANGES_REQUESTED** [review]
Just some details to improve, but anyway nice work!

**Pull request merged**